### PR TITLE
fix(config): Always output JSON format in `yarn config list`

### LINKED ISSUE #4496: Inconsistent config strings
Continuation from #4457 

**Do you want to request a *feature* or report a *bug*?**
Potential bug.

**What is the current behavior?**
Yarn config has different formatting for object keys that are computed / set vs defaults.

I'm not sure I understand if this is by design or not. If so, what is the use case?

screen shot 2017-09-17 at 5 15 06 pm

**What is the expected behavior?**
Keys should all be wrapped in single quotes or DEFAULTS should follow same formatting as `registry`, `lastUpdateCheck`, and user configured values do

**Please mention your node.js, yarn and operating system version.**
Yarn: 1.0.2
Node: 8.1.4

### PR DESCRIPTION
**Pull request by sreeramjayan**
**Summary**

Fixes #4496.

Used `JSON.stringify` to ensure the output of yarn config list displays JSON.

**Test plan**

New integration test to validate JSON output.

### CODE REVIEW

**Comment by arcanis** [comment]
Can you share a pic of before/after?

File: src/cli/commands/config.js [hunk 1]
```diff
@@ -83,10 +83,10 @@ export const {run, setFlags, examples} = buildSubCommands('config', {
     }
 
     reporter.info(reporter.lang('configYarn'));
-    reporter.inspect(config.registries.yarn.config);
+    reporter.inspect(JSON.stringify(config.registries.yarn.config, null, 2));
```
Comment (BYK) [inline src/cli/commands/config.js]: I think you should use `reporter.log` at this point with the `{force: true}` option.

**Comment by BYK** [comment]
Hey, thanks a lot for the PR!

A test case would be great if you can add it. A simple integration test in `__tests__/integration.js` would be sufficient to make sure the output is always valid JSON.

Also having a proper title for the PR would be great. We use the following format:
```
fix(config): Always output JSON format in `yarn config list`

**Summary**

Fixes #4496. Use `JSON.serialize` instead of `reporter.inspect` to ensure the output of the config object is always a valid JSON string.

**Test plan**

Added new integration test that checks JSON output.
```

**Comment by BYK** [comment]
@sreeramjayan let me know if you need any help with the tests!

**Comment by sreeramjayan** [comment]
@BYK I did make the requested changes but am facing an issue coming up with the test case.
https://github.com/sreeramjayan/yarn/commit/462afb2de70c90a85ba892eaa423c643068ef788

The issue seems to be that the output of running yarn config list isn't JSON since we get the results of the yarn and npm config.
https://github.com/sreeramjayan/yarn/blob/462afb2de70c90a85ba892eaa423c643068ef788/__tests__/integration.js#L420
 I was wondering how the output can be validated to be JSON.

**Comment by BYK** [comment]
Sorry for the late response and thanks a lot for the updates!

>The issue seems to be that the output of running yarn config list isn't JSON since we get the results of the yarn and npm config.

I think you can get a "pure" JSON string if you run `yarn --silent config list` ;)

**Comment by sreeramjayan** [comment]
@arcanis Here are the before and after screen shots. 
Before:
before

After:
after

As you mentioned the formatting is off when the string is serialized.

**Comment by sreeramjayan** [comment]
@arcanis Here are the before and after screen shots. 
Before:
before


After:
after


As you mentioned the formatting is off when the string is serialized.

**Comment by arcanis** [comment]
Hi! I'm really grateful for the work you put in this PR, but unfortunately the project moved forward since you submitted it and I fear it can be merged in this current state. I'm going to close it, but please feel free to check if the problem still exists, and to rebase your work on top of the current master and open a new PR if it's the case. Again, sorry we didn't got around merging it sooner!

**Pull request closed without merging**